code/game/objects/objs.dm

```diff
@@ -94,6 +93,10 @@
 	STOP_PROCESSING(SSobj, src)
 	return 0
 
+/obj/get_acid_delay()
+	. = ..()
+	if(density)
+		return 4 SECONDS
```

Comment:
if(density)
   return 4 seconds
return ..()

Reply:
use what BM has posted, we don't need to call parent if its dense, so we check that first